.github/workflows/create-class-diagram.yml

```diff
@@ -0,0 +1,51 @@
+---
+  name: Create class diagram
+
+  on:
+    push:
+      branches:
+        - main
+
+    workflow_call:
+
+  jobs:
+    create-class-diagram:
+      name: Create class diagram
+      runs-on: ubuntu-latest
+
+      steps:
+        - name: Check out repository
+          uses: actions/checkout@v4
+          with:
+            ref: ${{ github.event.pull_request.head.ref }}
+
+        - name: Install uv
+          uses: astral-sh/setup-uv@v5
+          with:
+            version: "0.5.28"  # install a specific version of uv
```

Comment:
why pin this version?

Reply:
As it was considered as best practice from uv: https://docs.astral.sh/uv/guides/integration/github/#installation
Also it is in a line with the locked uv version.

But of course, this may be part of a discussion :)